pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -360,6 +360,92 @@ func TestPriorityQueue_Delete(t *testing.T) {
 	}
 }
 
+func BenchmarkMoveAllToActiveOrBackoffQueue(b *testing.B) {
+	tests := []struct {
+		name                 string
+		podsInUnschedulableQ int
+		moveEvent            string
+	}{
+		{
+			name:      "baseline",
+			moveEvent: UnschedulableTimeout,
+		},
+		{
+			name:      "worst",
+			moveEvent: NodeAdd,
+		},
+		{
+			name: "random",
+			// leave "moveEvent" unspecified
+		},
+	}
+
+	podTemplates := []*v1.Pod{
+		highPriorityPodInfo.Pod, highPriNominatedPodInfo.Pod,
+		medPriorityPodInfo.Pod, unschedulablePodInfo.Pod,
+	}
+	events := []string{NodeAdd, NodeTaintChange, NodeAllocatableChange, NodeConditionChange, NodeLabelChange, PodAdd,
+		PvcAdd, PvcUpdate, PvAdd, PvUpdate, StorageClassAdd, CSINodeAdd, CSINodeUpdate}
+	pluginNum := 20
+	var plugins []string
+	// Mimic that we have 20 plugins loaded in runtime.
+	for i := 0; i < pluginNum; i++ {
+		plugins = append(plugins, fmt.Sprintf("fake-plugin-%v", i))
+	}
+
+	for _, tt := range tests {
+		for _, podsInUnschedulableQ := range []int{1000, 5000} {
+			b.Run(fmt.Sprintf("%v-%v", tt.name, podsInUnschedulableQ), func(b *testing.B) {
+				for i := 0; i < b.N; i++ {
+					b.StopTimer()
+					c := clock.NewFakeClock(time.Now())
+
+					m := make(map[framework.ClusterEvent]sets.String)
+					if tt.moveEvent != "" {
+						// Simulate that all plugins register the NodeAdd event.
+						m[clusterEventReg[NodeAdd]] = sets.NewString(plugins...)
+					} else {
+						// Random case.
```

Comment:
Try to describe what how is this distribution. I'm in reading this correctly, it's something like:

one event is registered for all plugins, the next event is registered for every other plugin, the next event is registered for every three plugins, and so on.

Reply:
Yes, added a comment:

```
// - All plugins registered for events[0]
// - 1/2 of plugins registered for events[1]
// - 1/3 of plugins registered for events[2]
// - ...
```